api/src/wfl/sink.clj

```diff
@@ -352,3 +371,10 @@
 (defoverload stage/done?  datarepo-sink-type datarepo-sink-done?)
 
 (defoverload util/to-edn datarepo-sink-type datarepo-sink-to-edn)
+
+;; reitit http coercion specs for a sink
+;; Recall s/or doesn't work (https://github.com/metosin/reitit/issues/494)
+(s/def ::sink
+  #(condp = (:name %)
+     terra-workspace-sink-name (s/valid? ::terra-workspace-sink %)
+     datarepo-sink-name        (s/valid? ::terra-datarepo-sink %)))
```

Comment:
Sadly when this fails we don't get very useful error messages as it won't tell you what's wrong with your request. All it'll say is your request doesn't match a workspace sink or a datarepo sink request. That said, the errors we return currently are truly dreadful.
I created https://broadinstitute.atlassian.net/browse/GH-1355 a while ago to address the shortcomings of these coercion errors.